tests/random_lax_test.py

```diff
@@ -365,6 +366,47 @@ def testCategorical(self, p, axis, dtype, sample_shape):
         pmf = lambda x: np.where(x < len(p), p[np.minimum(len(p) - 1, x)], 0.0)
         self._CheckChiSquared(samples, pmf=pmf)
 
+  @jtu.sample_product(
+    [
+      {
+        "logits_shape": logits_shape,
+        "axis": axis,
+      }
+      for logits_shape in [(2,), (3,), (2, 3), (2, 5), (2, 3, 5)]
+      for axis in range(-len(logits_shape), len(logits_shape))
+    ],
+    batch_shape=[(), (2,), (3,), (2, 3), (2, 5), (2, 3, 5), None],
+  )
+  def testCategoricalWithoutReplacement(self, logits_shape, axis, batch_shape):
+    key = random.key(0)
+
+    key, subkey = random.split(key)
+    logits = random.normal(subkey, logits_shape)
+
+    if batch_shape is None:
+      output = random.categorical(key, logits, axis, None, False)
+      assert output.shape == util.tuple_delete(logits_shape, axis, allow_neg=True)
+
+    else:
+      shape = batch_shape + util.tuple_delete(logits_shape, axis, allow_neg=True)
+
+      if math.prod(batch_shape) <= logits_shape[axis]:
+        output = random.categorical(key, logits, axis, shape, False)
+        assert output.shape == shape
+
+        flat = output.reshape(batch_shape + (-1,))
+
+        # flat = output.reshape((-1,) + util.tuple_delete(logits.shape, axis, allow_neg=True))
+
+        assert jax.vmap(all_unique, [-1])(flat).all()
+      else:
+        with self.assertRaisesRegex(
+          ValueError,
+          "k argument to top_k must be no larger than minor dimension",
```

Comment:
This is a pretty indirect error – could we detect this condition within `categorical` and raise something that's more actionable in that context?